cmd/imagec/docker.go

```diff
@@ -183,45 +185,89 @@ func FetchImageBlob(options ImageCOptions, image *ImageWithMeta) error {
 	// Open the file so that we can use it as a io.Reader for sha256 calculation
 	in, err := os.Open(string(imageFileName))
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer in.Close()
 
-	h := sha256.New()
+	done := make(chan bool)
+	errs := make(chan error)
+	defer close(done)
+	defer close(errs)
+
+	r, w := io.Pipe()
+	t := io.TeeReader(in, w)
+
+	// validate the sha256 sum of the compressed data
+	go func() {
+		defer w.Close()
+		h := sha256.New()
+		progress.Update(po, image.String(), "Verifying Checksum")
+		if _, cerr := io.Copy(h, t); cerr != nil {
```

Comment:
This looks sufficiently complicated that a nice comment of what is wired to what reader/writer would be appreciated.  

The single letter variables is making the logic pretty difficult to understand at first pass.  Please make things a bit easier to read if you can.  `piper, pipew, tr, xferSum`, etc.